substantial/types.py

```diff
@@ -101,26 +101,23 @@ class ValueEval:
     timeout: Union[float, None]
     retry_strategy: Union[RetryStrategy, None]
     compensate_fn: Optional[Callable[[], Any]]
+    compensation_stack: List[Callable[[], Any]]
     max_compensation_attempts: int = 3
 
     async def exec(
         self,
         ctx,
         save_id,
         counter,
-        compensation_stack: Optional[List[Callable[[], Any]]] = None,
     ) -> Any:
-        if compensation_stack is None:
-            compensation_stack = []
-
         strategy = self.retry_strategy or RetryStrategy(
             max_retries=3, initial_backoff_interval=0, max_backoff_interval=10
         )
 
         try:
             # if there is compessation, we need to add it to the stack
             if self.compensate_fn:
-                compensation_stack.append(self.compensate_fn)
+                self.compensation_stack.append(self.compensate_fn)
 
```

Comment:
Isn't a ref to the stack already available on ctx?

Reply:
Sorry, you're right!, anything else? I think this pr is done ??